src/triage/component/catwalk/storage.py

```diff
@@ -120,6 +127,26 @@ def delete(self):
     def open(self, *args, **kwargs):
         return self.client.open(self.path, *args, **kwargs)
 
+    @contextmanager
+    def local_fs_path(self):
+        """Expose a temporary filesystem path.
+
+        The path is prepopulated with the contents of the existing storage.
+        The path can be opened, read from, written to, and is automatically synced to S3 when the
+        context manager goes out of scope.
+
+        Yields: (str) a full local filepath that syncs with the base S3 storage
+        """
+        with tempfile.NamedTemporaryFile() as tf:
+            if self.exists():
+                with self.open('rb') as fd:
+                    shutil.copyfileobj(fd, tf)
+                tf.seek(0)
+            yield tf.name
+            tf.seek(0)
+            with self.open('wb') as fd:
+                shutil.copyfileobj(tf, fd)
```

Comment:
This is cool, though I'm sure it would be best to avoid :smile: 

I'm guessing we _had_ to pass along a local file system path, rather than a file descriptor? :confused:

Reply:
Yeah. It's very annoying. I could see this being useful for HDF5/S3 too, although it wouldn't use this interface.